lib/elixir/lib/stream/reducers.ex

```diff
@@ -25,7 +25,7 @@ defmodule Stream.Reducers do
     end
 
     after_fun = fn {acc_buffer, acc_count} ->
-      if leftover == :discard or acc_count == 0 do
+      if leftover == :discard or acc_count == 0 or (step > count and count == acc_count) do
```

Comment:
I think `count == acc_count` is a problem, you can test with:
```
assert Enum.chunk_every([1, 2, 3, 4, 5, 6, 7], 2, 4, []) == [[1, 2], [5, 6]]
```

Reply:
Thanks @ggcampinho I will return to this after finishing my working hours 👍